tests/unit/test_openstack.py

```diff
@@ -127,3 +272,37 @@ async def test_init_provider(self):
     async def test_provision(self):
         provider = OpenStackProvider()
         await provider.init(image_names=[])
+
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize("x", network_data, ids=[x[0] for x in network_data])
+    async def test_network_picking_validation(self, x):
+        _, hosts, pools, availability, exp_nets = x
+        reqs = deepcopy(hosts)
+        if availability:
+            del self.mock_neutron
+            self.mock_neutron = Mock()
+            self.mock_neutron.init_api = AsyncMock(return_value=True)
+            self.mock_neutron.network.list = AsyncMock(return_value=self.networks)
+            self.mock_neutron.ip.list = AsyncMock(return_value=availability)
+
+            self.mock_neutron_class = Mock(return_value=self.mock_neutron)
+            self.neutron_patcher = patch(
+                "mrack.providers.openstack.NeutronClient", new=self.mock_neutron_class
+            )
+            self.neutron_patcher.start()
```

Comment:
@miskopo thanks for help with this works like a charm.
![mpolovkarolls](https://media.giphy.com/media/3ohs4kI2X9r7O8ZtoA/giphy.gif)

Reply:
Always happy to help,  @Tiboris